packages/lit-dev-server/src/middleware/fake-github-middleware.ts

```diff
@@ -0,0 +1,133 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import type Koa from 'koa';
+
+/**
+ * Options for fakeGitHubMiddleware.
+ */
+export interface FakeGitHubMiddlewareOptions {
+  /**
+   * GitHub OAuth App client ID.
+   *
+   * In reality, this is generated when a GitHub OAuth App is first created. For
+   * this fake, it can be any string.
+   */
+  clientId: string;
+
+  /**
+   * GitHub OAuth app secret.
+   *
+   * In reality, this is generated from the GitHub OAuth App settings screen.
+   * For this fake, it can be any string.
+   */
+  clientSecret: string;
+
+  /**
+   * URL to redirect to after app authentication or installation.
+   *
+   * In reality, this is configured in the GitHub OAuth App settings screen. For
+   * this fake, it can be any URL.
+   */
+  redirectUrl: string;
+}
+
+/**
+ * Koa middleware that simulates various GitHub APIs and authentication screens
+ * used by the lit.dev Playground.
+ */
+export const fakeGitHubMiddleware = (
+  options: FakeGitHubMiddlewareOptions
+): Koa.Middleware => {
+  let fake = new FakeGitHub(options);
+  return async (ctx, next) => {
+    if (ctx.path === '/reset') {
```

Comment:
Is this a github endpoint or a testing convenience? Clarifying this will be under `/playground/signin/reset`?

Reply:
Just a testing convenience. Added a note to the `reset` method.

It will be under `http://localhost:5417/reset` -- all of these endpoints are on a separate server, simulating the GitHub server.